packages/vite/src/node/server/index.ts

```diff
@@ -219,6 +219,13 @@ export interface ViteDevServer {
    * and hmr state.
    */
   moduleGraph: ModuleGraph
+  /**
+   * A promise which will resolve once a reload is complete. A reload is
+   * pending as a result of optimizing missing dependencies.
+   * 
+   * When no reload is pending it will be null.
```

Comment:
I think we need to note here that the `pendingReload` field might be refreshed. That is, if the user visits a new route, and new dependencies are discovered, it may be re-assigned with a new `Promise`.